src/App.tsx

```diff
@@ -2,24 +1,44 @@
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [visiblePerPage, setVisiblePerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(0);
+  const pages: string[][] = Array.from(
+    { length: Math.ceil(items.length / visiblePerPage) },
+    (_, i) =>
+      items.slice(i * visiblePerPage, i * visiblePerPage + visiblePerPage),
+  );
+  const first = pages[currentPage].at(0);
+  const last = pages[currentPage].at(-1);
+  const from = first ? first.split(' ')[1] : 0;
+  const to = last ? last.split(' ')[1] : 0;
```

Comment:
The way you're getting the range of items currently being displayed is a bit convoluted. Consider storing the range in the state instead of calculating it from the current items.